strutil/env.go

```diff
@@ -0,0 +1,253 @@
+		}
+		env.entries[key] = value
+	}
+	return env, nil
+}
+
+// OSEnvironment returns the environment of the calling process.
+func OSEnvironment() (*Environment, error) {
+	return ParseRawEnvironment(os.Environ())
+}
+
+// Transform programmatically replaces all keys and values.
+//
+// If multiple keys are transformed into the same key the value of the last
+// (lexicographically) key is used. If the transformed key is empty then the
+// corresponding entry is removed.
+func (env *Environment) Transform(tr func(key, value string) (string, string)) {
+	newEntries := make(map[string]string, len(env.entries))
+	keys := make([]string, 0, len(env.entries))
+	for key := range env.entries {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		newKey, newValue := tr(key, env.entries[key])
+		if newKey != "" {
+			newEntries[newKey] = newValue
+		}
+	}
+	env.entries = newEntries
+}
+
+// Get returns the value of a given environment variable.
+func (env *Environment) Get(key string) string {
+	return env.entries[key]
+}
+
+// Contains returns true if given environment variable is set.
+func (env *Environment) Contains(key string) bool {
+	_, ok := env.entries[key]
+	return ok
+}
+
+// Del removes the given environment variable.
+func (env *Environment) Del(key string) {
+	delete(env.entries, key)
+}
+
+// Set adds or replaces the given environment variable.
+func (env *Environment) Set(key, value string) {
+	if env.entries == nil {
+		env.entries = make(map[string]string)
+	}
+	env.entries[key] = value
+}
+
+// RawEnvironment returns the raw equivalent of the given environment.
+// The returned environment is sorted lexicographically by variable name.
+func (env *Environment) RawEnvironment() RawEnvironment {
+	raw := make([]string, 0, len(env.entries))
+	keys := make([]string, 0, len(env.entries))
+	for key := range env.entries {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		raw = append(raw, fmt.Sprintf("%s=%s", key, env.entries[key]))
+	}
+	return raw
+}
+
+// EnvironmentDelta describes alterations to an environment.
+//
+// Differential environment can refer to existing entries by using shell-like
+// syntax $KEY or ${KEY}. Entries inside an environment delta are ordered.
+type EnvironmentDelta struct {
+	OrderedMap
+}
+
+// NewEnvironmentDelta returns a new environment delta comprised of given pairs.
+func NewEnvironmentDelta(pairs ...string) *EnvironmentDelta {
+	return &EnvironmentDelta{OrderedMap: *NewOrderedMap(pairs...)}
+}
+
+// Copy returns a copy of the environment delta.
+func (delta *EnvironmentDelta) Copy() *EnvironmentDelta {
+	return &EnvironmentDelta{OrderedMap: *delta.OrderedMap.Copy()}
+}
+
+// Merge combines two environment deltas.
+//
+// Clashing environment variables are overwritten and the value from the
+// "other" delta prevails.
+func (delta *EnvironmentDelta) Merge(other *EnvironmentDelta) {
```

Comment:
because of  https://github.com/snapcore/snapd/pull/8242/files#r390825566  Merge is probably not something we want, instead Apply* should take multiple of these (name to be clarified)

Reply:
Merge is gone. I'll tweak Apply next.